helm/charts/text-generation-inference/templates/service.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "text-generation-inference.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "text-generation-inference.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
```

Comment:
Is the port for metrics the same as the port for requests ?

Reply:
I believe they are, you need to call /generate at least once to get the metrics populated.
```
❯ curl -s -H "Content-type: application/json"  -k  https://startcoder.[redacted]/generate -d @/tmp/foo.json | jq

{
  "generated_text": "\nimport sys\nimport time\n\nfrom. import config\nfrom. import utils\nfrom."
}
❯ curl -sk https://startcoder.[redacted]/metrics | head
# TYPE tgi_request_success counter
tgi_request_success 3

# TYPE tgi_batch_inference_success counter
tgi_batch_inference_success{method="prefill"} 3
tgi_batch_inference_success{method="decode"} 57

# TYPE tgi_request_count counter
tgi_request_count 3
```